cg/meta/clean/demultiplexed_flow_cells.py

```diff
@@ -209,3 +215,53 @@ def remove_failed_flow_cell(self):
         self.remove_from_demultiplexed_runs()
         if self.files_exist_in_housekeeper and self.is_correctly_named:
             self.remove_files_from_housekeeper()
+
+    def archive_sample_sheet(self):
+        """Archives a sample sheet to /home/proj/production/sample_sheets and adds it to
+        Housekeeper with an appropriate tag"""
+        LOG.info("Archiving sample sheet for flow cell %s", self.run_name)
+
+        original_sample_sheet: Path = self.path / "SampleSheet.csv"
+        archived_sample_sheet: Path = self.sample_sheets_dir / self.run_name / "SampleSheet.csv"
+
+        try:
+            os.makedirs(self.sample_sheets_dir / self.run_name)
+            LOG.info("Sample sheet directory created for flow cell directory %s", self.run_name)
+        except FileExistsError:
+            LOG.info(
+                "Sample sheet directory for flow cell directory %s already exists!",
+                self.run_name,
+            )
+        try:
+            shutil.move(original_sample_sheet, archived_sample_sheet)
+            self.add_sample_sheet_to_housekeeper(str(archived_sample_sheet))
+            self.hk.commit()
+        except FileNotFoundError:
+            LOG.warning("No sample sheet found in flow cell directory %s!", self.run_name)
+            shutil.rmtree(self.sample_sheets_dir / self.run_name, ignore_errors=True)
+
+    def add_sample_sheet_to_housekeeper(self, sample_sheet_path: str):
```

Comment:
```suggestion
    def add_sample_sheet_to_housekeeper(self, sample_sheet_path: Path):
```
Otherwise the variable name `sample_sheet_path` is very misleadiing and I think you feed this archived_sample_sheet which is a path. If housekeeper needs a string (seems strange but still) make it a string in the very last step.